src/acl/ParameterizedNode.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_ACL_PARAMETERIZEDNODE_H
+#define SQUID_SRC_ACL_PARAMETERIZEDNODE_H
+
+#include "acl/Acl.h"
+#include "acl/Data.h"
+#include "base/Assure.h"
+
+#include <memory>
+
+namespace Acl
+{
+
+/// An ACL that manages configured parameters using a given Parameters class P.
+/// That Parameters class must support an ACLData<> or equivalent API.
+template <class P>
+class ParameterizedNode: public ACL
+{
+public:
+    using Parameters = P;
+
```

Comment:
Please just use the template definition to name the type instead of a separate sub-type.

```suggestion
/// An ACL that manages configured parameters using a given Parameters class.
/// That Parameters class must support an ACLData<> or equivalent API.
template <class Parameters>
class ParameterizedNode: public ACL
{
public:

```

Reply:
> Please just use the template definition to name the type instead of a separate sub-type.

Why? The current approach is better because it gives derived classes and external users a named type they can use. If we remove that alias as suggested, then derived classes would have to add duplicated aliases of their own (if they can guess the underlying type correctly), while some external users would have no good way of guessing the type at all.

For example, imagine writing a template function `foo(T &acl)` that accepts any ParameterizedNode acl object. How would that function get to the parameters type if we remove the alias? Today, that function can just use T::Parameters.

FWIW, this aliasing of template parameter types is standard practice, for similar [reasons](https://stackoverflow.com/questions/63104551/why-alias-a-template-parameter-in-c-standard-containers).

Reply:
> I do not think we have any code actually in Squid that needs that ability.

In this PR, there is at least one declaration in another class that already uses that ability:

```C++
FinalizedParameterizedNode(TypeName typeName, Parameters * const params)
```

The only way to avoid the above usage is to convert that simple constructor into a _template_ that accepts any `params` type even though the class itself needs a rather specific type. Such a conversion will make code more complex and compiler errors about mismatching ACLData types even more difficult to decipher (because the mismatch will happen at a lower code level). IMO, this "standard practice" `using` declaration is an acceptable price to pay for avoiding those complications (and it may get more reuse in the future).